.cbmc-batch/include/make_common_data_structures.h

```diff
@@ -25,3 +29,16 @@ void ensure_md_context_has_allocated_members(struct aws_cryptosdk_md_context *ct
 
 /* Allocates the members of the context and ensures that internal pointers are pointing to the correct objects. */
 void ensure_sig_ctx_has_allocated_members(struct aws_cryptosdk_sig_ctx *ctx);
+
+/**
+ * This function creates a valid alg properties data structure by
+ * calling the initializer in cipher.h. This is the only valid way to
+ * construct an alg_props structure, and that is why
+ * [aws_cryptosdk_alg_properties_is_valid] checks whether the pointer
+ * of alg_props is the same as the one returned after calling the
+ * [aws_cryptosdk_alg_props] initializer.
+ */
+void ensure_alg_properties_is_allocated(struct aws_cryptosdk_alg_properties **alg_props);
```

Comment:
I'm a little confused by this comment. What does it mean for the pointer to be the same? Looking below, it looks like the alg_props returns a singleton for each algorithm type, and we are checking that this is so.  This should be clearly documented in the relevant ESDK header if it's an expected invariant of the system.

Reply:
If I remember correctly this is the only allowed way to construct alg_props. Could someone from the ESDK team confirm?